README.md

```diff
@@ -209,27 +209,64 @@ Authorization in `root:redhat-appstudio` and `root:redhat-hacbs` workspaces in C
+
+$ ./hack/setup-monitoring.sh oauth2-secret grafana-oauth2-proxy $GRAFANA_GITHUB_CLIENT_ID $GRAFANA_GITHUB_CLIENT_SECRET $GRAFANA_GITHUB_COOKIE_SECRET
+```
+
+The `PROMETHEUS_GITHUB_CLIENT_ID`/`PROMETHEUS_GITHUB_CLIENT_SECRET` and `GRAFANA_GITHUB_CLIENT_ID`/`GRAFANA_GITHUB_CLIENT_SECRET` value pairs must match an existing "OAuth application" on GitHub - see [OAuth apps](https://github.com/organizations/redhat-appstudio-sre/settings/applications) in the [Red Hat Appstudio SRE organization](https://github.com/organizations/redhat-appstudio-sre). The `PROMETHEUS_GITHUB_COOKIE_SECRET` and `GRAFANA_GITHUB_COOKIE_SECRET` can be generated using the [following instructions](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#generating-a-cookie-secret).
+
+
+#### Grafana Datasources
+
+Grafana datasources contain the connection settings to the Prometheus instances. These datasources are stored in secrets in the `appstudio-workload-monitoring` of the **Grafana cluster**.
+
+The Prometheus endpoints used by Grafana are protected by an OAuth proxy running as a sidecar container and which checks that the incoming requests contain a valid token. A token is valid if it belongs to a service account of the **Prometheus cluster** and which has the RBAC permission to "get namespaces". This can be obtained with the `cluster-monitoring-view` cluster role.
+
+```
+$ ./hack/setup-monitoring.sh grafana-datasource-secret $DATASOURCE_NAME $PROMETHEUS_URL $GRAFANA_TOKEN
+```
+
+`DATASOURCE_NAME` is the name of the secret itself, as well as the base name of the YAML file it contains. Grafana supports datasources from multiple secrets, so their names must be unique.
+
+`PROMETHEUS_URL` is obtained from the route created for Prometheus in the `openshift-monitoring` and `appstudio-workload-monitoring` namespaces:
 
-```yaml
-   apiVersion: v1
-   kind: Secret
-   metadata:
-     namespace: appstudio-workload-monitoring
-     name: prometheus-proxy-config
-     labels:
-       provider: appstudio-kcp-workload
-   data:
-     client-id: <github-oauth-client-id-for-prometheus>
-     client-secret: <github-oauth-secret-for-prometheus>
-     cookie-secret: <github-oauth-cookie-secret-for-prometheus>
-   type: Opaque
 ```
+$ PROMETHEUS_URL=`oc get route/prometheus-k8s -n openshift-monitoring -o json | jq -r '.status.ingress[0].host'`
+
+$ PROMETHEUS_URL=`oc get route/prometheus-oauth -n appstudio-workload-monitoring -o json | jq -r '.status.ingress[0].host'`
+```
+
+`GRAFANA_TOKEN` is obtained by requesting a token for the `grafana-oauth` service account in the **Prometheus cluster**:
+```
+$ GRAFANA_TOKEN=`oc create token grafana-oauth -n appstudio-workload-monitoring`
+```
+
+The `grafana-oauth` service account is created by `.components/monitoring/base/configure-prometheus.yaml` along with a binding to the `cluster-monitoring-view` cluster role. 
```

Comment:
How do we get Argo CD to apply this configuration?

Reply:
I'm not sure I understand your question: this `grafana-oauth` service account and its associated cluster role binding are defined in `components/monitoring/base/configure-prometheus.yaml`, so it going to be managed by ArgoCD, isn't it?

Reply:
I haven't gotten far enough to see if it works.  Since it's in the base configuration, I think this prometheus configuration will be applied to every cluster where you run the bootstrap script.  Let's verify that and then I would add that statement here to avoid confusion.  Otherwise people might think they need to `kubectl apply` it.

Reply:
ok, makes sense. Yes, it should be applied by ArgoCD on every cluster (as opposed to Grafana on a single cluster), but we should mention that for the sake of clarity. I'll add that once we have `./components/monitoring` in place.